core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -207,6 +211,24 @@ void execute(Terminal terminal, String pluginId, boolean isBatch, Map<String, St
             Files.createDirectory(env.pluginsFile());
         }
 
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
How will this react in tests? We would end up creating a bunch of hooks for temp files?